### PR TITLE
PMA-69: fix board redirecting

### PR DESCRIPTION
**Pull request by cptShell**

### CODE REVIEW

File: src/components/board/board.tsx [hunk 1]
```diff
@@ -45,6 +51,12 @@ export const Board: FC = () => {
 
   useEffect(() => {
     updateColumns();
+
+    return () => {
+      const resetFunc = (): void => setColumns(null);
+
+      dispatch(boardActions.reset(resetFunc));
+    };
```
Comment (liza-veis) [inline src/components/board/board.tsx]: Can we do this in this way?
```suggestion
    updateColumns();

    return () => {
      dispatch(boardActions.reset());
    };
```

and then in useEffect
```
useEffect(() => {
  setColumns(board?.columns ?? []);
}, [board?.columns]);
```

File: src/store/board/actions.ts [hunk 2]
```diff
@@ -56,3 +56,10 @@ export const removeBoard = createAsyncThunk<string, string, AsyncThunkConfig>(
     return payload;
   },
 );
+
+export const reset = createAsyncThunk<void, () => void, AsyncThunkConfig>(
```
Comment (liza-veis) [inline src/store/board/actions.ts]: use createAction instead of createAsyncThunk

**Review by liza-veis — APPROVED** [review]

File: src/components/board/board.tsx (hunk 1 above)
Comment (cptShell) [reply src/components/board/board.tsx]: I dont need another updating useEffect because i have updateColumns

File: src/components/board/board.tsx [hunk 3]
```diff
@@ -35,7 +38,10 @@ export const Board: FC = () => {
 
   const updateColumns = async (): Promise<void> => {
     if (boardId) {
-      dispatch(boardActions.getById(boardId));
+      const data = await dispatch(boardActions.getById(boardId)).unwrap();
+      const currentBoard = data as FullBoardDto;
```
Comment (liza-veis) [inline src/components/board/board.tsx]: this logic should be in action

**Review by AelxRmoan — APPROVED** [review]

**Review by Nazeroth — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**